msdk-id-sirius/src/main/java/SiriusIdentificationMethod.java

```diff
@@ -0,0 +1,349 @@
+        int atomicNumber = isotope.getAtomicNumber();
+        final Element element = periodicTable.get(atomicNumber);
+        int min = range.getIsotopeCountMin(isotope);
+        int max = range.getIsotopeCountMax(isotope);
+
+        constraints.setLowerbound(element, min);
+        if (max!= maxNumberOfOneElements) constraints.setUpperbound(element, max);
+      }
+
+      logger.debug("ConstraintsGenerator finished");
+      return constraints;
+    }
+
+  }
+
+  public double getParentMass() {
+    return parentMass;
+  }
+
+  public IonType getIonization() {
+    return ion;
+  }
+
+  public List<MsSpectrum> getMsSpectra() {
+    return ms1;
+  }
+
+  public List<MsSpectrum> getMs2Spectra() {
+    return ms2;
+  }
+
+  public int getNumberOfCandidates() {
+    return numberOfCandidates;
+  }
+
+  /**
+   *
+   * <p> Method for reading custom MS files (just 2 columns with mz & intentsity values </p>
+   * Does similar to mgf parser functionality
+   */
+  public static MsSpectrum readCustomMsFile(File file, String delimeter)
+      throws IOException, MSDKRuntimeException {
+    logger.info("Started reading {}", file.getName());
+
+    Scanner sc = new Scanner(file);
+    ArrayList<String> strings = new ArrayList<>();
+    while (sc.hasNext()) {
+      strings.add(sc.nextLine());
+    }
+    sc.close();
+
+    int size = strings.size();
+    double mz[] = new double[size];
+    float intensity[] = new float[size];
+
+    int index = 0;
+    for (String s : strings) {
+      String[] splitted = s.split(delimeter);
+      if (splitted.length == 2) {
+        mz[index] = Double.parseDouble(splitted[0]);
+        intensity[index++] = Float.parseFloat(splitted[1]);
+      } else {
+        throw new MSDKRuntimeException("Incorrect spectrum structure");
+      }
+    }
+
+    MsSpectrumType type = SpectrumTypeDetectionAlgorithm.detectSpectrumType(mz, intensity, size);
+    logger.info("Finished reading {}", file.getName());
+
+    return new SimpleMsSpectrum(mz, intensity, size, type);
+  }
+
+  /**
+   * <p> Method for processing spectra by Sirius module </p>
+   * Transformation of MSDK data structures into Sirius structures and processing by Sirius
+   * Method is left to be protected for test coverage
+   */
+  protected List<IdentificationResult> siriusProcessSpectra() throws MSDKException {
+    Spectrum<Peak> siriusMs2;
+    MsSpectrum msdkSpectrumMs2 = ms2.remove(0);
+
+    double mz[] = msdkSpectrumMs2.getMzValues();
+    double intensity[] = LocalArrayUtil.convertToDoubles(msdkSpectrumMs2.getIntensityValues());
+
+    siriusMs2 = sirius.wrapSpectrum(mz, intensity);
+    String ionization = ion.getName();
+    PrecursorIonType precursor = sirius.getPrecursorIonType(ionization);
+
+    /* MutableMs2Experiment allows to specify additional fields and it is exactly what comes from .getMs2Experiment */
+    MutableMs2Experiment experiment = (MutableMs2Experiment) sirius.getMs2Experiment(parentMass, precursor, null, siriusMs2);
```

Comment:
Here you should not assume that sirius.getMs2Experiment() always returns MutableMs2Experiment. Maybe future Sirius versions will return a different object. What is the actual return type of the function?
You should first check using the instanceof operator, before doing the cast to MutableMs2Experiment.